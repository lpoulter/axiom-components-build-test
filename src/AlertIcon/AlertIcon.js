import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';
import Icon from '../Icon/Icon';
import './AlertIcon.css';

export const ALERT_ICON_NAME_MAP = {
  error: 'warning',
  info: 'information',
  success: 'tick',
  warning: 'warning',
};

const SIZE_MAP = {
  small: '1rem',
  medium: '2rem',
  large: '3rem',
};

export default class AlertIcon extends Component {
  static propTypes = {
    /** Size of the indicator and icon */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    /** Style of the indicator */
    style: PropTypes.oneOf(['primary', 'secondary', 'subtle']),
    /** Color of the indicator and icon */
    type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
  };

  static defaultProps = {
    size: 'small',
    style: 'primary',
    type: 'info',
  };

  render() {
    const { size, style, type, ...rest } = this.props;
    const classes = classnames(
      'ax-alert-icon',
      `ax-alert-icon--${style}`,
      `ax-alert-icon--${size}`,
      `ax-alert-icon--${type}`
    );

    return (
      <Base { ...rest } className={ classes }>
        <Icon name={ ALERT_ICON_NAME_MAP[type] } size={ SIZE_MAP[size] } />
      </Base>
    );
  }
}
