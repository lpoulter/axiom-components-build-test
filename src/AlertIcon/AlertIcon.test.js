import React from 'react';
import renderer from 'react-test-renderer';
import AlertIcon from './AlertIcon';

const getComponent = (props = {}) =>
  renderer.create(
    <AlertIcon name="twitter" { ...props } />
  );

describe('AlertIcon', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with color', () => {
    ['success', 'warning', 'error', 'info', 'subtle'].forEach((color) => {
      it(color, () => {
        const component = getComponent({ color });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with size', () => {
    ['small', 'medium', 'large'].forEach((size) => {
      it(size, () => {
        const component = getComponent({ size });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with a primary or secondary style', () => {
    ['primary', 'secondary'].forEach((style) => {
      it(style, () => {
        const component = getComponent({ style });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
