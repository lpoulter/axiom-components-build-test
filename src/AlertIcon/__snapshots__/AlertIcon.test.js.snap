// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertIcon renders with a primary or secondary style primary 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with a primary or secondary style secondary 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--secondary ax-alert-icon--small ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with color error 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  color="error"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with color info 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  color="info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with color subtle 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  color="subtle"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with color success 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  color="success"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with color warning 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  color="warning"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with defaultProps 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with size large 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--large ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "3rem",
        "width": "3rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with size medium 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--medium ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "2rem",
        "width": "2rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;

exports[`AlertIcon renders with size small 1`] = `
<div
  className="ax-alert-icon ax-alert-icon--primary ax-alert-icon--small ax-alert-icon--info"
  name="twitter"
>
  <svg
    className="ax-icon ax-icon--information"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
    style={
      Object {
        "height": "1rem",
        "width": "1rem",
      }
    }
    viewBox="0 0 16 16"
  />
</div>
`;
